Added support for move to parent in main view.

rdnetto opened the pull request:
Fixed #388.

jonas commented on src/tig.c:
```diff
@@ -266,7 +266,11 @@ view_driver(struct view *view, enum request request)
 	case REQ_PARENT:
-		report("Moving to parent is not supported by the %s view", view->name);
+		if (! strcmp(view->name, "main")) {
+			move_view(view, request);
```
The changes to support `REQ_PARENT` should go in `src/main.c` which is all the main view specific code. There's a `main_request` method where the `case REQ_PARENT` and related `move_view` changes in this patch can exist.

rdnetto commented:
I've moved the code into `main_request()`, and modified it to reuse an existing method instead of duplicating the scrolling logic.

I also added a special case for treating HEAD as the parent of Unstaged Changes.

jonas commented:
Thanks, I ended up massaging the code a bit myself and also added a test and a NEWS entry.

chenkaie commented:
That would be nicer if support 2 direction movement, a really intuitive musle memory when I doing the movement :)
- `,`              : move to parent
- `SHIFT + ,` : move to child
